### ISSUE TITLE
Many null bytes sent with UUID in TLS enrollment

### ISSUE THREAD

**Issue by zwass** [facebook/osquery#2601]
```
I1007 18:30:34.424208 12634 tls.cpp:205] TLS/HTTPS POST request to URI: https://localhost:8080/enroll
{"enroll_secret":"","host_identifier":"1D9CE8AA-8029-43EB-9D85-350ED0DEBB24\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000...(thousands more)"}
```

I can repro this consistently on an osquery 2.0.0 binary installation on Centos7.1 and Ubuntu14. It does not occur on a build from master on OS X.

The following config file is sufficient to repro:
```
{
  "options": {
    "enroll_tls_endpoint": "/enroll",
    "tls_hostname": "localhost:8080",
    "verbose": true,
    "tls_dump": true,
    "host_identifier": "uuid",
    "disable_distributed": false
    }
}
```

First reported by @qudu in the #kolide channel on osquery slack (on osquery2.0.0/Centos7.2).

**Comment by theopolis** [comment]
Interesting, what is the output of:
```
$ osqueryi --line "select uuid, length(uuid) from system_info"
        uuid = ba94e859-9115-4365-8048-3c4c8cf0c20f
length(uuid) = 36
```

**Comment by zwass** [comment]
Ah, yes, I meant to include the information that the select returns the expected results:

```
vagrant@vagrant-ubuntu-trusty-64:~$ osqueryi --line "select uuid, length(uuid) from system_info"
        uuid = fdb3aabd-dc5b-41e2-bead-8e19afda1d63
length(uuid) = 36
```

**Comment by theopolis** [comment]
@zwass if your output is 36 characters that suggests there aren't `null`s appended?

**Comment by zwass** [comment]
Fixed by #2616
